core/services/postgres/transaction.go

```diff
@@ -3,11 +3,28 @@ package postgres
 import (
 	"context"
 	"database/sql"
+	"fmt"
+	"time"
 
 	"github.com/jinzhu/gorm"
 	"github.com/pkg/errors"
 )
 
+// NOTE: In an ideal world the timeouts below would be set to something sane in
+// the postgres configuration by the user. Since we do not live in an ideal
+// world, it is necessary to override them here.
+//
+// They cannot easily be set at a session level due to how Go's connection
+// pooling works.
```

Comment:
We would have to create a custom driver to automatically set these on every session and I don't think that's worth it at this time.